src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -826,6 +829,33 @@ void ep_rt_aot_create_activity_id (uint8_t *activity_id, uint32_t activity_id_le
 #endif
 }
 
+ep_rt_thread_handle_t ep_rt_aot_thread_get_handle (void)
+{
+    ThreadStore::AttachCurrentThread();
+    Thread* pThread = ThreadStore::GetCurrentThread();
+    return pThread;
+}
+
+ep_rt_thread_id_t ep_rt_aot_thread_get_id (ep_rt_thread_handle_t thread_handle)
+{
+    return thread_handle->GetPalThreadIdForLogging();
```

Comment:
The `GetPalThreadIdForLogging` name seems like it is not intended for 'normal' use. Is adding the usage here breaking some assumption now?

Reply:
`EEThreadId `classes seems to be doing the right thing to set its first field (`pthread_self `and `GetCurrentThreadId`), but doesn't expose a getter (reason for not using for 'normal' use?) `GetPalThreadIdForLogging `is using this field directly to get the value. The other option is to use `PalGetCurrentOSThreadId `(save at `get_or_create `time and return in this method).

Reply:
So I'd guess the reason it is just for logging is that `pthread_self` returns an opaque `pthread_t`. And `GetPalThreadIdForLogging` just returns it as a numeric value, so using it outside of logging hides the fact that it is not just a number and you can't always do things like just check equality by comparing the number.

Maybe that works for the purposes of what you are trying to do in this PR, but it seems like something that should be revisited/fixed.

Reply:
Filed #89650 to track this